src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -770,6 +770,37 @@ internal override bool IsExplicitDefinitionOfNoPiaLocalType
             }
         }
 
+        private void CheckPresenceOfTypeIdentifierAttribute()
+        {
+            // We want this function to be as cheap as possible, it is called for every top level type
+            // and we don't want to bind attributes attached to the declaration unless there is a chance
+            // that one of them is TypeIdentifier attribute.
+            ImmutableArray<SyntaxList<AttributeListSyntax>> attributeLists = GetAttributeDeclarations();
+
+            foreach (SyntaxList<AttributeListSyntax> list in attributeLists)
+            {
+                var syntaxTree = list.Node.SyntaxTree;
+                HashSet<string> quickCheckSet = this.DeclaringCompilation.GetBinderFactory(list.Node.SyntaxTree).GetBinder(list.Node).QuickTypeIdentifierAttributeCheckSet;
+
+                foreach (AttributeListSyntax attrList in list)
+                {
+                    foreach (AttributeSyntax attr in attrList.Attributes)
+                    {
+                        string name = attr.Name.GetUnqualifiedName().Identifier.ValueText;
```

Comment:
Would it be sufficient to search for attributes named `"TypeIdentifier"` or `"TypeIdentifierAttribute"`?

Reply:
> Would it be sufficient to search for attributes named "TypeIdentifier" or "TypeIdentifierAttribute", without creating the QuickTypeIdentifierAttributeCheckSet?

That would not take aliases into account, and therefore could produce inaccurate result.